### DISCUSSION TITLE
Middleware in phalcon v4

### DISCUSSION THREAD

**Discussion by zain-akram-1** [phalcon/cphalcon#16082] [Q&A] [answered]
How we can create a middleware in Phalcon v4?

**Comment by niden** [comment, marked as answer]
This is the only middleware type that Phalcon supports for the moment

https://docs.phalcon.io/4.0/en/application-micro#middleware

Using the micro application you can easily add what you need and it will be processed as you need it. There is also a sample for this repository:

https://github.com/phalcon/rest-api

In the future the HTTP layer will be rewritten to fully support middleware.
